lib/system/web/src/animationframeloop.rs

```diff
@@ -0,0 +1,60 @@
+use super::request_animation_frame;
+use wasm_bindgen::prelude::Closure;
+use std::rc::Rc;
+use std::cell::RefCell;
+
+
+// ============================
+// === Animation Frame Data ===
+// ============================
+
+struct AnimationFrameData {
+    run : bool
+}
+
+pub struct AnimationFrameLoop {
+    forget : bool,
+    data   : Rc<RefCell<AnimationFrameData>>
+}
+
+// ============================
+// === Animation Frame Loop ===
+// ============================
+
+impl AnimationFrameLoop {
+    pub fn new(mut func : Box<dyn FnMut()>) -> Self {
+        let nop_func    = Box::new(|| ()) as Box<dyn FnMut()>;
+        let nop_closure = Closure::wrap(nop_func);
+        let callback    = Rc::new(RefCell::new(nop_closure));
+
+        let run  = true;
+        let data = Rc::new(RefCell::new(AnimationFrameData { run }));
+
+        let callback_clone = callback.clone();
+        let data_clone     = data.clone();
+
+        *callback.borrow_mut() = Closure::wrap(Box::new(move || {
```

Comment:
We expects that function will be called only once from js, so we can use [once_for_js](https://docs.rs/wasm-bindgen/0.2.55/wasm_bindgen/closure/struct.Closure.html#method.once_into_js)

Reply:
`Closure::once_into_js` returns a `JsValue` which cannot be used with `request_animation_frame`

Reply:
Right. Then there is another function you may consider: `Closure::once`